src/openms/source/KERNEL/MSSpectrum.cpp

```diff
@@ -480,7 +480,8 @@ namespace OpenMS
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wfloat-equal"
     return std::operator==(*this, rhs) &&
-           RangeManagerType::operator==(rhs) &&
+           // do not compare range manager because it depends on manual calls to updateRanges
+           // RangeManagerType::operator==(rhs) && 
```

Comment:
i would strongly vote against this.
If you kick out RangeManager, you'd have to kick out SpectrumSettings as well... and where do you stop?
If  `expA.getMinIntensity() != expB.getMinIntensity()`, why should `expA == expB` be true?
If the ranges are not equal, the objects are not equal (and the bug is somewhere else).

Reply:
My argument here would be that Range is not part of the state of the object (or only for technical reasons) as it needs to be manually updated. SpectrumSettings is. But ok let's look at other solutions.

> If expA.getMinIntensity() != expB.getMinIntensity(), why should expA == expB be true?
In chromatograms we also don't compare the name but base equality on the content (peaks and meta data).

The issue particularly arises when we compare the same spectrum from OnDisc, Cached (memory mapped), and in memory spectra.
I can change these parts to call updateRanges... maybe it is not a speed bottleneck.

Reply:
mhh.. thinking about this more: if two Spectra are identical, apart from outdated ranges, one would need to call .updateRanges just to make this comparison succeed.
So yeah, lets remove Ranges from `operator==` for MSSpec, MSChrom and MSExp, but this needs to be documented in the code and in the doxygen of the methods to prevent accidental future 'changes'. Also a test would be good.